File: app/src/main/java/peter/util/searcher/tab/Tab.java
package peter.util.searcher.tab;


import peter.util.searcher.db.dao.TabData;

/**
 *
 * Created by peter on 2016/11/17.
 */

public interface Tab {

    String LOCAL_SCHEMA = "local://";
    String URL_HOME = LOCAL_SCHEMA + "home";
    String ACTION_NEW_WINDOW = "new_window";

    void loadUrl(TabData bean);

    String getUrl();

    String getSearchWord();

    int getPageNo();

    void reload();

    boolean canGoBack();

    void goBack();

    boolean canGoForward();

    void goForward();

    String getTitle();

    String getHost();

}

File: app/src/main/java/peter/util/searcher/tab/Tab.kt
package peter.util.searcher.tab


import peter.util.searcher.bean.Bean

/**
 *
 * Created by peter on 2016/11/17.
 */

interface Tab {

    companion object {
        val LOCAL_SCHEMA = "local://"
        val URL_HOME = LOCAL_SCHEMA + "home"
        val ACTION_NEW_WINDOW = "new_window"
    }

    fun getUrl(): String?

    fun getSearchWord():String?

    fun getPageNo(): Int?

    fun getTitle():String?

    fun getHost():String?

    fun loadUrl(bean: Bean)

    fun reload()

    fun canGoBack(): Boolean

    fun goBack()

    fun canGoForward(): Boolean

    fun goForward()

    fun onCreate()

    fun onPause()

    fun onResume()

    fun onDestroy()

}
